### ISSUE TITLE
Solidus 3.4.0.dev is not compatible with SolidusFrontend 3.4.0.dev

### ISSUE THREAD

**Issue by gsmendoza** [solidusio/solidus#4878]
**Solidus Version**:

master (3.4.0.dev)

**To Reproduce**

Run `bin/sandbox --frontend=classic`. 

**Current behavior**

You would get the following message:

> Your bundle is locked to solidus_frontend (3.3.0) from https://github.com/solidusio/solidus_frontend.git (at
master@99378d5), but that version can no longer be found in that source. That means the author of
solidus_frontend (3.3.0) has removed it. You'll need to update your bundle to a version other than
solidus_frontend (3.3.0) that hasn't been removed in order to install.

**Expected behavior**

The sandbox generator would be able to add the solidus_frontend master version without any errrors.

**Additional context**

I think this is the code that is producing the message:

```ruby
# core/lib/generators/solidus/install/app_templates/frontend/classic.rb

  # `solidus_frontend` is not sourced from rubygems if the solidus gem was not.
  github_solidus_frontend = '--github solidusio/solidus_frontend' unless solidus.nil? || solidus.source.is_a?(Bundler::Source::Rubygems)

  bundle_command("add solidus_frontend #{github_solidus_frontend}")
```

When generating the sandbox app, Solidus tries to source solidus_frontend from github. However, Solidus currently depends on SolidusFrontend 3.3.0.alpha, while the master branch version of SolidusFrontend has been updated to 3.4.0.dev.

The issue is also causing The SolidusBraintree CI to fail. From https://app.circleci.com/pipelines/github/solidusio/solidus_braintree/199/workflows/df45f354-6e9d-474c-8cd9-6c33a4a1b1cc/jobs/501/parallel-runs/0/steps/0-125:

```
bundle lock --update
Fetching https://github.com/solidusio/solidus_frontend.git
Fetching https://github.com/solidusio/solidus.git
Fetching gem metadata from https://rubygems.org/..........
Resolving dependencies...
Bundler could not find compatible versions for gem "solidus_frontend":
  In Gemfile:
    solidus_frontend

    solidus was resolved to 3.3.0, which depends on
      solidus_frontend (~> 3.3.0.alpha)

Could not find gem 'solidus_frontend (~> 3.3.0.alpha)' in
https://github.com/solidusio/solidus_frontend.git (at master@99378d5).

The source contains the following gems matching 'solidus_frontend':
  * solidus_frontend-3.4.0.dev

Exited with code exit status 6
CircleCI received exit code 6
```

## Planned solution

* [x] Relase SolidusFrontend 3.4.0.dev.
* [x] Revert https://github.com/solidusio/solidus/pull/4885.
* [ ] Update the SolidusFrontend dependency to 3.4.0.dev.

**Comment by gsmendoza** [comment]
Unfortunately, SolidusBraintree CI is still failing (see https://app.circleci.com/pipelines/github/solidusio/solidus_braintree/212/workflows/72ccf982-f67f-41b5-a342-13d4ab2f88af), so https://github.com/solidusio/solidus/pull/4885 didn't fix the issue for SolidusBraintree.

I think the test I ran yesterday is producing a false positive. https://app.circleci.com/pipelines/github/solidusio/solidus_braintree/213/workflows/d273a631-710c-4cd6-9855-0a1eaf2b898b is still passing right now. In particular, if you look at the lint-code job, it is able to generate the Gemfile.lock file with `bundle lock --update` (https://app.circleci.com/pipelines/github/solidusio/solidus_braintree/213/workflows/d273a631-710c-4cd6-9855-0a1eaf2b898b/jobs/542/parallel-runs/0/steps/0-106). However, if I ssh to the CI box and run `bundle lock --update`, it is  not able to fetch solidus_frontend:

```
circleci@41f8766f626a:~/project$ bundle lock --update
Fetching https://github.com/solidusio/solidus_frontend.git
Fetching https://github.com/nebulab/solidus.git
Fetching gem metadata from https://rubygems.org/..........
Resolving dependencies...
Bundler could not find compatible versions for gem "solidus_frontend":
  In Gemfile:
    solidus_frontend

    solidus was resolved to 3.4.0.dev, which depends on
      solidus_frontend (~> 3.3.0.alpha)

Could not find gem 'solidus_frontend (~> 3.3.0.alpha)' in https://github.com/solidusio/solidus_frontend.git
(at master@99378d5).

The source contains the following gems matching 'solidus_frontend':
  * solidus_frontend-3.4.0.dev
```

**Comment by gsmendoza** [comment]
I figured out why my test was producing a false positive. In [https://github.com/solidusio/solidus_braintree/commit/12e4fdeeaca2822b4bded4e3e41556ed6f5cba84#diff-d09ea66f8227784ff439[…]0031d16c93d67e6283ab55fR9](https://github.com/solidusio/solidus_braintree/commit/12e4fdeeaca2822b4bded4e3e41556ed6f5cba84#diff-d09ea66f8227784ff4393d88a19836f321c915ae10031d16c93d67e6283ab55fR9), I assumed that the `SOLIDUS_BRANCH` environment variable would be nil for the master branch in CI. Instead `ENV['SOLIDUS_BRANCH']` is set to `"master"` in CI. As a result, the `gem 'solidus_frontend', github: 'solidusio/solidus_frontend' if branch == MASTER_BRANCH` line was not executed, since `MASTER_BRANCH` was set to my development branch..
